kratos/geometries/line_2d_2.h

```diff
@@ -830,85 +855,92 @@ class Line2D2 : public Geometry<TPointType>
     }
 
     /**
-     * Returns whether given arbitrary point is inside the Geometry and the respective
+     * Returns whether given arbitrary point is inside the Geometry and the respective 
      * local point for the given global point
+     * @param rPoint: The point to be checked if is inside o note in global coordinates
+     * @param rResult: The local coordinates of the point
+     * @param Tolerance: The  tolerance that will be considered to check if the point is inside or not
+     * @return True if the point is inside, false otherwise
      */
-    virtual bool IsInside( const CoordinatesArrayType& rPoint, CoordinatesArrayType& rResult, const double Tolerance = std::numeric_limits<double>::epsilon() ) override
+    virtual bool IsInside( 
+        const CoordinatesArrayType& rPoint, 
+        CoordinatesArrayType& rResult, 
+        const double Tolerance = std::numeric_limits<double>::epsilon() 
+        ) override
     {
         PointLocalCoordinates( rResult, rPoint );
 
-        if ( (rResult[0] >= (-1.0 - Tolerance)) && (rResult[0] <= (1.0 + Tolerance)) )
+        if ( std::abs( rResult[0] ) <= (1.0 + Tolerance) )
         {
             return true;
         }
-        else
+        
+        return false;
+    }
+    
+    /**
+     * Returns whether given arbitrary point is inside, once this node has been projected if possible,
+     * the Geometry and the respective local point for the given global point
+     * @param rPoint: The point to be checked if is inside o note in global coordinates
+     * @param rResult: The local coordinates of the point
+     * @param Tolerance: The  tolerance that will be considered to check if the point is inside or not
+     * @return True if the point is inside, false otherwise
+     */
+    virtual bool IsInsideWhenProjected( 
+        const CoordinatesArrayType& rPoint, 
+        CoordinatesArrayType& rResult, 
+        const double Tolerance = std::numeric_limits<double>::epsilon(),
+        const array_1d<double,3> Direction = ZeroVector(3) 
+        ) override
+    {
+        PointLocalCoordinatesWhenProjected( rResult, rPoint, Direction );
```

Comment:
I think I need first to have  a clarification of what I need to do before starting to modify anything